contracts/mesh-provider/src/contract.rs

```diff
@@ -37,6 +37,7 @@ pub fn instantiate(
         slasher: None,
         lockup: deps.api.addr_validate(&msg.lockup)?,
         unbonding_period: msg.unbonding_period,
+        denom: msg.rewards_ibc_denom,
```

Comment:
Yea, It comes from the init msg, I did a hash function in the tests to quickly hash the ibc denom, which is easy to use and pass.
Also im not sure if we have hashing library exposed on cosmwasm, even if we do, it would probably be cheaper to just pass it to init.

Is there a way to "unhash" the ibc hash? because its weird to me that you might have ibc coins laying around that you forgot the channel you got them on (or just don't know what channel), and now you have no way of knowing where you need to send it to redeem the coins.
The only way is by trying to hash different channels and match what you need, or actually building a function that just loops over channel ids one by one and matches to your needed hash (not a bad idea actually)